core/src/main/java/org/bitcoinj/core/Wallet.java

```diff
@@ -3971,7 +3971,8 @@ public boolean canSignFor(Script script) {
     protected List<UTXO> getStoredOutputsFromUTXOProvider() throws UTXOProviderException {
         UTXOProvider utxoProvider = checkNotNull(vUTXOProvider, "No UTXO provider has been set");
         List<UTXO> candidates = new ArrayList<UTXO>();
-        List<DeterministicKey> keys = getActiveKeyChain().getLeafKeys();
+        List<ECKey> keys = getImportedKeys();
+        keys.addAll(getActiveKeyChain().getLeafKeys());
```

Comment:
@mikehearn From the API design perspective, do you think it's ok to modify the list we got from Wallet.getImportedKeys()? If so, should we perhaps document the allowed ownership takeover?